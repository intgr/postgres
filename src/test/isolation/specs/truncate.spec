setup
{
  CREATE TABLE foo (i int);
  INSERT INTO foo VALUES (1);
}

teardown
{
  DROP TABLE foo;
}

session "s1"
step "begin"
{
  BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
  -- Force snapshot open
  SELECT 1 AS foo FROM txid_current();
}
step "select"	{ SELECT * FROM foo; }
step "roll"		{ ROLLBACK; }

session "s2"
step "trunc"	{ TRUNCATE TABLE foo; }
