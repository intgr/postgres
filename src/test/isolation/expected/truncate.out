Parsed test spec with 2 sessions

starting permutation: begin select roll trunc
step begin: 
  BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
  -- Force snapshot open
  SELECT 1 AS foo FROM txid_current();

foo            

1              
step select: SELECT * FROM foo;
i              

1              
step roll: ROLLBACK;
step trunc: TRUNCATE TABLE foo;

starting permutation: begin select trunc roll
step begin: 
  BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
  -- Force snapshot open
  SELECT 1 AS foo FROM txid_current();

foo            

1              
step select: SELECT * FROM foo;
i              

1              
step trunc: TRUNCATE TABLE foo; <waiting ...>
step roll: ROLLBACK;
step trunc: <... completed>

starting permutation: begin trunc select roll
step begin: 
  BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
  -- Force snapshot open
  SELECT 1 AS foo FROM txid_current();

foo            

1              
step trunc: TRUNCATE TABLE foo;
step select: SELECT * FROM foo;
ERROR:  canceling statement due to conflict with TRUNCATE TABLE on foo
step roll: ROLLBACK;

starting permutation: trunc begin select roll
step trunc: TRUNCATE TABLE foo;
step begin: 
  BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
  -- Force snapshot open
  SELECT 1 AS foo FROM txid_current();

foo            

1              
step select: SELECT * FROM foo;
i              

step roll: ROLLBACK;
